File: homework-springboot/src/main/java/com/ibm/fsd/service/UserService.java
package com.ibm.fsd.service;

import com.ibm.fsd.domain.User;

import java.util.List;

/**
 * 用户管理Service
 */
public interface UserService {

    /**
     * 查询所有用户
     *
     * @return 用户列表
     */
    List<User> findAll();

    /**
     * 保存单个用户
     *
     * @param user 用户信息
     */
    void save(User user);

    /**
     * 保存用户集合
     *
     * @param list 用户信息列表
     */
    void save(List<User> list);

    /**
     * Find User by Login Name
     *
     * @return 用户
     */
    User findByLoginName(String loginName);

    boolean isLoginNameExist(String loginName);
}

File: springboot-ui/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpService } from './http.service';
import { User } from '../model/user.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private httpSvc: HttpService) { }

  /**
   * Add User
   * @param user  
   */
  userRegister(user: User): Observable<Boolean> {
    return this.httpSvc.post('/register',user).pipe();
  }

  /**
   * User Login
   * @param user 
   */
  userLogin(user: User) : Observable<Boolean> {
    return this.httpSvc.post('/login',user).pipe();
  }

  /**
   * User Update
   * @param user
   */
  userUpdate(user: User): Observable<Boolean> {
    return this.httpSvc.post('/update',user).pipe();
  }

  /**
   * Get User by login name
   * @param loginName
   */
  getUser(loginName: string): Observable<User> {
    var url = '/' + loginName;
    return this.httpSvc.get(url).pipe();
  }

}
